TCF4.py

```diff
@@ -0,0 +1,121 @@
+from Bio import SeqIO
+from Bio.SeqUtils import GC
+
+def filter_fastq(input_path: str, quality_threshold: int, output_filename="final_filtered.fastq",  gc_bounds=(40, 60), length_bounds=(50, 350)):
+    filename = input_path
+    records = SeqIO.parse(filename, "fastq")
+    ###quality filter
+    good_reads = (rec for rec in records if min(rec.letter_annotations["phred_quality"]) >= quality_threshold)
+    result_quality = SeqIO.write(good_reads, "good_quality.fastq", "fastq")
+    result_quality_GC = SeqIO.parse("good_quality.fastq", "fastq")
+    ###GC content filter
+    min_gc_content = gc_bounds[0]
+    max_gc_content = gc_bounds[1]
+    GC_quality_filt = []
+    
+    for sequence in result_quality_GC:
+        if min_gc_content <= GC(sequence.seq) <= max_gc_content:
+            GC_quality_filt.append(sequence)
+            
+    result_quality = SeqIO.write(GC_quality_filt, "good_quality_GC.fastq", "fastq")
+    result_quality_GC_length = SeqIO.parse("good_quality_GC.fastq", "fastq")
+    
+    ##length filter
+    filtered_GC_quality_length = []
+    
+    for sequence in result_quality_GC_length:
+        if len(sequence.seq) >= length_bounds[0] and len(sequence.seq) <= length_bounds[1]:
+            filtered_GC_quality_length.append(sequence)
+            
+    result_quality = SeqIO.write(filtered_GC_quality_length, output_filename, "fastq")
+    
+    print(result_quality)
```

Comment:
Логику последовательных проверок можно реализовать чуть короче, не создавая итератор для каждой новой проверки:

```suggestion
def filter_fastq_corr(input_path: str, quality_threshold: int, output_filename="final_filtered_corr.fastq",  gc_bounds=(40, 60), length_bounds=(50, 350)):
    if isinstance(gc_bounds, int):
        gc_bounds = (0, gc_bounds)
    min_gc_content, max_gc_content = gc_bounds
    length_min, length_max = length_bounds
    
    records = SeqIO.parse(input_path, "fastq")
    good_reads = []
    for rec in records:
        rec_len, rec_gc_content = len(rec.seq), GC(rec.seq)
        if (
            min(rec.letter_annotations["phred_quality"]) >= quality_threshold
            and max_gc_content >= rec_gc_content >= min_gc_content
            and length_max >= rec_len >= length_min
        ):
            good_reads.append(rec)
    result_quality = SeqIO.write(good_reads, output_filename, "fastq")

    print(result_quality)
```